### PR TITLE
Adding a scoped storage migration channel

### PR DESCRIPTION
**Pull request by Arthur-Milchior**

### CODE REVIEW

**Comment by lukstbit** [comment]
Compilation errors for tests.

**Review by lukstbit — APPROVED** [review]
Combined with squash merge the changes LGTM.
The second commit(lint complains about API level) and the third commit(the importance property is not defined) fail.

**Comment by BrayanDSO** [comment]
This has got some conflicts

File: AnkiDroid/src/main/java/com/ichi2/anki/NotificationChannels.kt [hunk 1]
```diff
@@ -38,25 +43,43 @@ object NotificationChannels {
             Channel.GLOBAL_REMINDERS -> res.getString(R.string.widget_minimum_cards_due_notification_ticker_title)
             Channel.DECK_REMINDERS -> res.getString(R.string.deck_conf_reminders)
             Channel.GENERAL -> res.getString(R.string.app_name)
+            Channel.SCOPED_STORAGE_MIGRATION -> res.getString(R.string.scoped_storage_channel_name)
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NotificationChannels.kt:46]:
```suggestion
            Channel.SCOPED_STORAGE_MIGRATION -> res.getString(R.string.scoped_storage_title)
```

File: AnkiDroid/src/main/res/values/02-strings.xml [hunk 2]
```diff
@@ -406,6 +406,7 @@
     <string name="scoped_storage_learn_more">Learn More</string>
     <string name="scoped_storage_postpone">Postpone</string>
     <string name="scoped_storage_migrate">Migrate</string>
+    <string name="scoped_storage_channel_name">AnkiDroid upgrade</string>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/02-strings.xml:409]:
```suggestion
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Apparently has some conflicts and *if the goal is to rebase merge) some issues with the commit stream on individual commits (https://github.com/ankidroid/Anki-Android/blob/main/tools/test-commits.sh is an easy way to test them, though it will melt your machine ;-) - I run it `/bin/nice -n 19 ./tools/test-commits.sh <hash>` personally)

My actual suggestions are just me continuing to act as police for the strings and push my personal vision of "Storage migration" as the phrase to use. I recognize what string to use is still up for discussion though

**Comment by Arthur-Milchior** [comment]
So, I changed the text, and rebased.

**Review by mikehardy — APPROVED** [review]
I fixed up the last 2 commits and pushed it out, pending CI, thanks!

**Review by mikehardy — CHANGES_REQUESTED** [review]
Grrr - an API availability lint check - must be fixed

**Comment by Arthur-Milchior** [comment]
I failed to understand why it was compiling locally. Until I saw I used `RequiresApi` instead of `TargetApi`.

**Review by mikehardy — APPROVED** [review]
And now it's not even a strings change :-)
Thanks for the final fixup

**Pull request merged**